src/io/flutter/run/FlutterLaunchMode.java

```diff
@@ -20,11 +22,21 @@
   RELEASE("release");
 
   private static final Key<FlutterLaunchMode> LAUNCH_MODE_KEY = Key.create("FlutterLaunchMode");
+  private static final Key<FlutterApp> APP_KEY = Key.create("FlutterApp");
 
   public static void addToEnvironment(ExecutionEnvironment env, FlutterLaunchMode mode) {
     env.putUserData(FlutterLaunchMode.LAUNCH_MODE_KEY, mode);
   }
 
+  public static void addToEnvironment(@NotNull ExecutionEnvironment env, @NotNull FlutterApp app) {
```

Comment:
These add / from env. methods should likely be moved to `FlutterApp` (to parallel `FlutterLaunchMode` and `RunMode`).

Reply:
Good idea; moved.